staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/lastappliedupdater_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager_test
+
+import (
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager"
+	"sigs.k8s.io/yaml"
+)
+
+func TestUpdateLastApplied(t *testing.T) {
+	f := NewTestFieldManager(schema.FromAPIVersionAndKind("apps/v1", "Deployment"))
+	f.fieldManager = fieldmanager.NewLastAppliedUpdater(
+		f.fieldManager,
+	)
+
+	appliedObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {
+        “apiVersion”: “apps/v1”,
+        “kind”: ”Deployment”,
+        “metadata”: {
+          “name”: “my-deployment”
+        },
+        “spec”: {
+          “replicas”: “3”,
+          “selector”: {
+            “matchLabels”: {
+              “app”: ”my-app”
+            }
+          }
+        }
+      }
+spec:
+  replicas: 100000
+  selector:
+    matchLabels:
+      app: my-app
+  template:
+    metadata:
+      labels:
+        app: my-app
+    spec:
+      containers:
+      - name: my-c
+        image: my-image
+`), &appliedObj.Object); err != nil {
+		t.Fatalf("error decoding YAML: %v", err)
+	}
+
+	if err := f.Apply(appliedObj, "fieldmanager_test_apply", false); err != nil {
+		t.Fatalf("failed to update object: %v", err)
+	}
+
+	accessor, err := meta.Accessor(f.liveObj)
+	if err != nil {
+		t.Fatalf("couldn't get accessor")
+	}
+
+	var annotations = accessor.GetAnnotations()
+	if annotations == nil {
+		t.Fatalf("could not get annotations")
+	}
+
+	lastApplied, ok := annotations[corev1.LastAppliedConfigAnnotation]
+	if !ok {
+		t.Fatalf("last applied annotation is empty")
+	}
+
+	expected := `
```

Comment:
I think that newline is failing your test.

Reply:
Hm I'm going to move this test somewhere else since the test field manager doesn't use the `merge`